## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [1]:
import numpy as np 
import pandas as pd

In [2]:
from statsmodels.stats.weightstats import ztest
import scipy.stats as stats

In [16]:
data_path = '../Модуль 6 Статистика/data.csv'
df = pd.read_csv(data_path)

In [14]:
data = df['data'].to_numpy()

In [17]:
df

Unnamed: 0       data
0            0  17.499453
1            1  19.662399
2            2   7.182823
3            3  29.841625
4            4   9.239386
..         ...        ...
95          95  26.244944
96          96  22.282832
97          97  18.697188
98          98  27.041189
99          99   5.938381

[100 rows x 2 columns]

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [6]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [7]:
n = df['data'].count()
n

100

In [8]:
avg_vyb = 20

In [9]:
t_stat, p_value = stats.ttest_1samp(df['data'], avg_vyb, alternative='greater')
t_stat, p_value

(-0.9953477389335053, 0.8390031080494006)

P = 0.83 > 0.05, не можемо відкинути нульову гіпотезу. Отже, начущо **підвищити** середній чек за останній місяць не вдалось

2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [18]:
data_path = '../Модуль 6 Статистика/data_2.csv'
df_serp = pd.read_csv(data_path)

In [19]:
df_serp

data
0     5.940159
1    29.189344
2    11.383502
3    15.434098
4    19.896752
..         ...
115  31.304916
116  15.822676
117  19.291394
118  19.036747
119  18.847899

[120 rows x 1 columns]

In [20]:
data.shape

(120,)

In [21]:
df = df.drop(columns=["Unnamed: 0"])

In [24]:
mean_df = df['data'].sum()/len(df['data'])
mean_df_serp = df_serp['data'].sum()/len(df_serp['data'])

In [25]:
mean_df, mean_df_serp

(19.37755319670441, 19.37322276512519)

In [26]:
ztest_Score, p_value = ztest(df['data'], df_serp['data'])

In [27]:
ztest_Score, p_value

(0.005246673456808504, 0.9958137794593356)

P = 0.996 > 0.05, не можемо відкинути нульову гіпотезу. Отже, середні значення чеків у липні та серпні статистично не відрізняються